### PR TITLE
Fixes #1457 gc for R

### LINKED ISSUE #1457: GC for R
Export GC function in OSPSuite.R

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

File: src/OSPSuite.Core/Services/GarbageCollectionTask.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace OSPSuite.Core.Services
+{
+   public static class GarbageCollectionTask
+   {
+      /// <summary>
+      ///    Force GC Collection compacting the Large Object Heap. Implementation very similar to what DotMemory
+      ///    is doing according to one of their post
+      /// </summary>
+      public static void ForceGC()
```
Comment (msevestre) [inline src/OSPSuite.Core/Services/GarbageCollectionTask.cs:12]: I think this is a better way of clearing the memory. Let see...

File: src/OSPSuite.Core/Workspace.cs [hunk 2]
```diff
@@ -48,7 +49,7 @@ public void UpdateJournalPathRelativeTo(string projectFileFullPath)
       protected void ReleaseMemory()
       {
          //This might free some C++ objects still  being held into memory
-         this.GCCollectAndCompact();
```
Comment (msevestre) [inline src/OSPSuite.Core/Workspace.cs:51]: Having it as extension method was stupid. This is better as a simple static method

File: src/OSPSuite.R/Api.cs [hunk 3]
```diff
@@ -65,6 +65,14 @@ public static void InitializeOnce(ApiConfig apiConfig)
 
       public static IDataImporterTask GetDataImporterTask() => resolveTask<IDataImporterTask>();
 
+      /// <summary>
+      /// Forces the Garbage collection 
+      /// </summary>
+      public static void ForceGC()
```
Comment (msevestre) [inline src/OSPSuite.R/Api.cs:71]: Availalbe for R

**Review by msevestre — COMMENTED** [review]
ready

File: src/OSPSuite.Core/Services/GarbageCollectionTask.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      {
+         GCSettings.LargeObjectHeapCompactionMode = GCLargeObjectHeapCompactionMode.CompactOnce;
+         for (int i = 0; i < 4; i++)
+         {
+            GC.Collect(2, GCCollectionMode.Forced, blocking: true);
+            GC.WaitForPendingFinalizers();
+         }
```
Comment (Yuri05) [inline src/OSPSuite.Core/Services/GarbageCollectionTask.cs:19]: where does the 4 in the GC-loop actually come from?

Comment (msevestre) [reply src/OSPSuite.Core/Services/GarbageCollectionTask.cs:19]: no idea. Empirical value apparently that mimics so behavior. There is always stuff left if you do it only 3 times or less

Comment (Yuri05) [reply src/OSPSuite.Core/Services/GarbageCollectionTask.cs:19]: hmm, ok :)

**Review by Yuri05 — APPROVED** [review]

File: src/OSPSuite.Core/Services/GarbageCollectionTask.cs (hunk 4 above)
Comment (msevestre) [reply src/OSPSuite.Core/Services/GarbageCollectionTask.cs:20]: I have added  a link to the discussion in code
https://dotnettools-support.jetbrains.com/hc/en-us/community/posts/360000088690-How-reproduce-DotMemory-s-Force-GC-button-s-behaviour-on-code-with-c-

**Pull request merged**